execution-core/src/lib.rs

```diff
@@ -29,19 +29,20 @@ pub use bls12_381_bls::{
     Signature as BlsSignature, APK as BlsAggPublicKey,
 };
 
-/// Secret key associated to a stake.
-pub type StakeSecretKey = BlsSecretKey;
-/// Public key associated to a stake.
-pub type StakePublicKey = BlsPublicKey;
-/// Signature associated with a stake.
-pub type StakeSignature = BlsSignature;
+/// Secret key associated to an account.
```

Comment:
The reason for this is that a Moonlight *account*'s key is the exact same as a stake/provisioner key. Given this equivalence, and given that a stake also belongs to an *account*, I personally think that the work "account" is far more appropriate than any of the other options discussed.

Reply:
> We use the bls-signature scheme to identify accounts both in the stake contract and in moonlight (and possibly in other contexts in the future), so I think the name is actually quite fitting.

I agree and also think it is a good idea to have it more generalised with the account prefix, especially with the possibility of its use in even more contexts in the future.

Reply:
Apparently my last comment con Accounts got lost... I'll try to rewrite it (and update it according to my latest understanding).

From what I can see, a BLSPubKey is not inherently associated with an account. For instance, if a user never transacts with Moonlight, it's key would never be an account. In this respect, I think Account is a specific case of what I would call a `UserKey` or something like that. Within Moonlight, the `UserKey` would be an `AccountKey`.
While in the Stake contract, the `UserKey` would be a `ProvisionerKey`.

I know understand that you see provisioners as accounts in the Stake contract. I generally tend to disagree, as I see an account as a (key, balance) tuple, while in Stake there are two quantities: the stake, and the reward.
Also, from an account you can usually withdraw any amount, but this is only true for the reward, while the stake can only be withdrawn in its totality.

In conclusion, I still disagree with using Account everywhere. BUT, if I can't convince you to change it, we should at least make it explicit that we consider provisioners as accounts in the Stake contract, and use the term stake only to refer to the `stake` value. For instance, `loaded_stake` or `load_or_create_stake` are not correct in this view.

Reply:
Today in another episode of "Why naming things is hard" :stuck_out_tongue_winking_eye:

I'm not sure I can keep everyone happy here.

There is the issue of separating rewards and stakes in the contract itself, which seems to have a consensus on it being a good thing. Only @herr-seppia could possibly disagree. I'm of the opinion this, and everything flowing from it, including the "is in the stake contract but not a provisioner" distinction, should be handled in a different PR/issue, since it would be unwise to further broaden the scope of this one.

There is also the aliasing of `Bls*` entities into `Account`. I have two options here:

1.) Rename everything back to `Bls*` and tackle an alias in the future. I know @Neotamandua and @moCello like it, but I'm also sensitive to @fed-franz 's objections, and since it's arguably out of scope for this PR in the first place, I don't have a problem with changing it.
2.) Keep as is and deal with the objections, and whatever change they may bring, conceptual or otherwise, in a different discussion/issue.

Personally, I would choose 2.) since there's more than enough good arguments in favor of the change, but still enough weight to the objections to warrant consideration. Plz react with the appropriate number (1 or 2) to vote. Comments are fine too.

Reply:
I'd go for 1, rename everything back to `Bls*`, as I think it is suboptimal to mix application concepts and technical concepts in naming of highly technical objects, which might also be used universally.

Reply:
> I'd go for 1, rename everything back to `Bls*`, as I think it is suboptimal to mix application concepts and technical concepts in naming of highly technical objects, which might also be used universally.

This is probably the best argument against `Account` so far imo.

Reply:
(Unsurprisingly) I vote for 1.
I also agree with @miloszm, and I think it aligns with what I'm trying to say.

Reply:
I have pushed the removal of the `Account*` aliases in favor of using just `Bls*`. Option 1. it is.